moveit_simple/include/moveit_simple/point_types.h

```diff
@@ -154,5 +160,72 @@ class CartTrajectoryPoint : public TrajectoryPoint
 private:
   Eigen::Affine3d pose_;
 };
+
+class CombinedTrajectoryPoint : public TrajectoryPoint
+{
+public:
+  enum PointPreference
+  {
+    CARTESIAN = 0,
+    JOINT = 1
+  };
+
+  CombinedTrajectoryPoint() : TrajectoryPoint("", 0.0, PointType::JOINT)
+  {
+  }
+
+  CombinedTrajectoryPoint(std::vector<double> &joint_point, const Eigen::Affine3d pose, double t, double tol,
+                          std::string name = std::string(), PointPreference type = PointPreference::JOINT,
+                          JointLockOptions options = JointLockOptions::LOCK_NONE, double timeout = 5.0)
+    : TrajectoryPoint(name, t, (type == PointPreference::JOINT) ? PointType::JOINT : PointType::CARTESIAN, options)
+    , options_(options)
+    , joint_point_(joint_point)
+    , pose_(pose)
+    , tol_(tol)
+  {
+  }
+
+  virtual ~CombinedTrajectoryPoint() { }
+
+  const JointLockOptions &jointLockOptions() const
+  {
+    return options_;
+  }
+
+  const Eigen::Affine3d &pose() const
+  {
+    return pose_;
+  }
+  
+  const std::vector<double> &jointPoint() const
+  {
+    return joint_point_;
+  }
+
+  const double &timeout() const
+  {
+    return timeout_;
+  }
+
+  std::string pointVecToString(const std::vector<double> &vec) const;
+
+protected:
+  
+  virtual std::unique_ptr<JointTrajectoryPoint>
+  toJointTrajPoint(const Robot &robot, double timeout, const std::vector<double> &seed,
+                   JointLockOptions options = JointLockOptions::LOCK_NONE) const;
+
+  virtual std::unique_ptr<CartTrajectoryPoint> toCartTrajPoint(const Robot &robot) const;
+
+  bool compareJointAndCart(const Robot &robot, double timeout) const;
+
+private:
+
```

Comment:
Why not utilize the Joint/Cart types here?  It seems like you are replicating the private variables of those types?

Reply:
To reduce redundant storage. Cart/Joint points have some of overlap, toJointTrajPoint and toCartTrajPoint already create points the sub-type.